src/features/components/molecules/protected-route/index.tsx

```diff
@@ -0,0 +1,21 @@
+import React from 'react'
+import PageNotFound from 'features/pages/404-error'
+import InitialLoader from 'components/elements/dot-loader'
+
+interface ProtectedRouteProps {
+    condition: boolean
+    params?: any
```

Comment:
it is not advisable to have `any` type. Can we pass the exact component with props instead so we don't need to create a separate `param` prop

ex:

```
  <ProtectedRoute
            is_loading={is_region_loading}
            component={<AsiansDigitalOptionsPage prop_a={1} prop_b={2} prop_c={3}  />}
            condition={is_row}
        />
```